src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -995,20 +1004,139 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_fillBody(JSC::JSGlob
+{
+    if (thisLength < valueLength + byteOffset)
+        return -1;
+    auto start = thisPtr + byteOffset;
+    auto end = thisPtr + thisLength;
+    auto it = static_cast<uint8_t*>(memmem(start, thisLength - byteOffset, valuePtr, valueLength));
+    if (it != NULL) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int lastIndexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    auto start = thisPtr;
+    auto end = thisPtr + std::min(thisLength, byteOffset + valueLength);
+    auto it = std::find_end(start, end, valuePtr, valuePtr + valueLength);
+    if (it != end) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int indexOf(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis, bool last)
 {
     auto& vm = JSC::getVM(lexicalGlobalObject);
-    return JSC::JSValue::encode(jsUndefined());
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    if (callFrame->argumentCount() < 1) {
+        throwVMError(lexicalGlobalObject, scope, createNotEnoughArgumentsError(lexicalGlobalObject));
+        return JSValue::encode(jsUndefined());
+    }
+
+    auto value = callFrame->uncheckedArgument(0);
+    WebCore::BufferEncodingType encoding = WebCore::BufferEncodingType::utf8;
+
+    int length = static_cast<int>(castedThis->byteLength());
+    const uint8_t* typedVector = castedThis->typedVector();
+
+    int byteOffset = last ? length - 1 : 0;
+
+    if (callFrame->argumentCount() > 1) {
+        auto byteOffset_ = callFrame->uncheckedArgument(1).toNumber(lexicalGlobalObject);
+        if (std::isnan(byteOffset_) || std::isinf(byteOffset_)) {
+            byteOffset = last ? length - 1 : 0;
+        } else if (byteOffset_ < 0) {
+            byteOffset = length + static_cast<int>(byteOffset_);
+        } else {
+            byteOffset = static_cast<int>(byteOffset_);
+        }
+
+        if (last) {
+            if (byteOffset < 0) {
+                return -1;
+            } else if (byteOffset > length - 1) {
+                byteOffset = length - 1;
+            }
+        } else {
+            if (byteOffset <= 0) {
+                byteOffset = 0;
+            } else if (byteOffset > length - 1) {
+                return -1;
+            }
+        }
+
+        if (callFrame->argumentCount() > 2) {
+          std::optional<BufferEncodingType> encoded = parseEnumeration<BufferEncodingType>(*lexicalGlobalObject, callFrame->argument(2));
+          if (!encoded) {
+              throwTypeError(lexicalGlobalObject, scope, "Invalid encoding"_s);
+              return JSC::JSValue::encode(jsUndefined());
+          }
+
+          encoding = encoded.value();
+        }
+    }
+
+    if (value.isString()) {
+        auto* str = value.toString(lexicalGlobalObject);
+        JSC::EncodedJSValue encodedBuffer = constructFromEncoding(lexicalGlobalObject, str, encoding);
```

Comment:
Instead of allocating an entire JS object here

We should be able to only allocate a string if we have the right overloads

As an additional optimization, we can stack allocate a 1024 character buffer and then use it if there are less than say 128 characters 

I can add this though if you want to move on

Reply:
it would be great if you can help with this :)